Emulation Support for ZynqU

chvamshi-xilinx opened the pull request:
Hi All,

This change contains emulation support of ZynqU platform (zcu102ng).

Thanks
Vamshi Krishna

gbuildx commented:
Can one of the admins verify this patch?

mamin506 commented on src/runtime_src/driver/zynq/hw_em/pllauncher_defines.h:
```diff
@@ -0,0 +1,236 @@
+ * under the License.
+ */
+
+/*
```
@sonals Should we have MIT license here?

chvamshi-xilinx replied:
Added same lincense which hardware driver contains. We are including hardware driver file in emulation file

mamin506 commented on src/runtime_src/driver/zynq/sw_em/pllauncher_defines.h:
```diff
@@ -0,0 +1,236 @@
+ * under the License.
+ */
+
+/*
```
MIT license again. @sonals  Please review this license also.

chvamshi-xilinx replied:
Added same lincense which hardware driver contains. We are including hardware driver file in emulation file

mamin506 commented on src/runtime_src/tools/scripts/petalinux.sh:
```diff
@@ -152,7 +152,11 @@ petalinux-build
+if [ ! -d ${XRT_REPO_DIR}/src/platform/zcu102ng/src/boot ] ; then
+  mkdir -p ${XRT_REPO_DIR}/src/platform/zcu102ng/src/boot
+fi
+
```
This change is not necessary. "mkdir -p" will return without error if directory existed.

chvamshi-xilinx commented:
Hi Min,

I made the requested changes to this code and updated the license.

I am using Apache License as we are including Zynq hardware driver in emulation driver. I think we should use same License which you are using.

Thanks
Vamshi Krishna

chvamshi-xilinx commented:
Thanks Nik. I have added directory existence check in petalinux.sh but i didn't observe that you are already using mkdir with -p option. I reverted the change. There is no other change in petalinux.sh

chvamshi-xilinx commented:
Following up with Qemu team to get the sources of DTBs/Elf files.

I will make required changes once i get a response from Qemu Team.

chvamshi-xilinx commented:
Removed the cosim's dtb's dependency. Working with Qemu team to resolve static elf file issue.
